<launch>
    <arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="config" default="baseline" doc="Specify a config."/>
  	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />

    <arg name="map" doc="Specify the map you wants, must be specified but user"/>
    <arg name="IP" default="192.168.1.17" doc="The IP address that the server laptop connected. The default was set in config file. See 00-infrastructure/duckietown/config/baseline/tcp_communication"/>

    <arg name="GUI" default="true" doc="Show GUI or not"/>

    <arg name="same_master" default="false" doc="If we connect all machine to the same master. (laptop)" />

    <group unless="$(arg same_master)">
      <include file="$(find auto_localization)/launch/subfserver.launch">
        <arg name="veh" value="$(arg veh)"/>
        <arg name="config" value="$(arg config)"/>
        <arg name="param_file_name" value="$(arg param_file_name)"/>
        <arg name="IP" value="$(arg IP)"/>
        <arg name="map" value="$(arg map)"/>
      </include>
      <remap from="local_poses" to="subfserver_easy/local_poses" />
    </group>

    <include file="$(find auto_localization)/launch/absolute_from_relative_position.launch">
      <arg name="veh" value="$(arg veh)"/>
      <arg name="map" value="$(arg map)"/>
      <arg name="output" value="$(arg map)"/>
      <arg name="same_master" value="$(arg same_master)"/>
    </include>

    <remap from="bot_global_poses" to="absolute_from_relative_position/bot_global_poses" />
    <include file="$(find auto_localization)/launch/pose_optimization.launch">
      <arg name="veh" value="$(arg veh)"/>
      <arg name="output" value="$(arg map)"/>
    </include>

    <remap from="bot_global_poses_optimized" to="pose_optimization/bot_global_poses_optimized" />
    <include if="$(arg GUI)" file="$(find auto_localization)/launch/map_description.launch">
      <arg name="veh" value="$(arg veh)"/>
      <arg name="map" value="$(arg map)"/>
    </include>

</launch>
